core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -165,15 +167,17 @@ public boolean delete(Supplier<JournalContext> journalContext, AlluxioURI uri) {
     try (LockResource r = new LockResource(mWriteLock)) {
       if (mState.getMountTable().containsKey(path)) {
         // check if the path contains another nested mount point
-        for (String mountPath : mState.getMountTable().keySet()) {
-          try {
-            if (PathUtils.hasPrefix(mountPath, path) && (!path.equals(mountPath))) {
-              LOG.warn("The path to unmount {} contains another nested mountpoint {}",
-                  path, mountPath);
-              return false;
+        if (checkNestedMount) {
```

Comment:
Why do we need to introduce this flag?

Reply:
So basically if user execute this `updateMount`command on a mount point that contains a nested mount. The below check would fail because for a normal unmount we do not want to unmount if a nested mount point exist. In our case for updateMount command we do not want to limit the user to only be able to update mount point that does not have a nested mount, so this logic is disable in the `updatedMount` case. It is safe in this case because the `updateMount` command will keep the inode tree under the mount point so the nested mount point can still exist after the command.

Reply:
This seems quite tricky in cases of partial success, if there is a partial mount, will we even be able to delete the data (alluxio only) in the nested mount?

Reply:
yeah at the end of the `updateMount` function it will capture any exception and delete the subtree which also removes any nested mount along the way.